src/map.cpp

```diff
@@ -1320,34 +1318,33 @@ vehicle *map::displace_vehicle( tripoint &p, const tripoint &dp )
 
     // record every passenger inside
     std::vector<int> psg_parts = veh->boarded_parts();
-    std::vector<player *> psgs;
+    std::vector<Creature *> psgs;
+    const tripoint old_veh_pos = veh->global_pos3();
     for( auto &prt : psg_parts ) {
+        auto psg = veh->get_passenger( prt );
+        const tripoint part_pos = old_veh_pos + veh->parts[prt].precalc[0];
         psgs.push_back( veh->get_passenger( prt ) );
```

Comment:
Why is this calling `get_passenger` again, you already know who is there (it's `psg`).